app/models/spree/retail/shopify/generate_refund_order.rb

```diff
@@ -4,24 +4,31 @@ module Shopify
       class GenerateRefundOrder
         def initialize(shopify_refund)
           @shopify_refund = shopify_refund
+          @order = find_order(shopify_refund)
         end
 
         def process
-          order = find_order(shopify_refund)
-
-          return_items = []
-          shopify_refund.refund_line_items.each do |rli|
-            inventory_unit = find_inventory_by_shopify_variant_id(order, rli.line_item.variant_id)
-            return_items << create_return_item(inventory_unit)
-          end
-
           create_return_authorization(order, return_items)
           customer_return = create_customer_return(return_items)
           create_reimbursement(customer_return)
         end
 
         private
 
+        attr_reader :shopify_refund, :order
+
+        def return_items
+          @return_items ||= begin
```

Comment:
Yummy! Love it. Remind me to show you `inject` sometime to make examples like this even cleaner.